crypto/cms/cms_sd.c

```diff
@@ -267,11 +276,13 @@ CMS_SignerInfo *CMS_add1_signer(CMS_ContentInfo *cms,
     if (!si)
         goto merr;
     /* Call for side-effect of computing hash and caching extensions */
+    X509_set0_libctx(signer, ctx->libctx, ctx->propq);
```

Comment:
Should the caller be responsible for doing this here?

Reply:
The caller is the app. Are you sure you want every app to have to set this themselves? More than likely it will fall over in a fecth and they wont have any idea why it happened.

Reply:
Removed